script/GameObjects/Elements/Core/Children.lua

```diff
@@ -30,8 +30,15 @@ function Entity:addChildren ()
 end
 
 function Entity:getChildren ()
+  -- Return non-component children
   assert(self.children)
-  return self.children
+  local realChildren = {}
+  for _, child in ipairs(self.children) do
+    if child:hasTrackable() and child:isTrackable() then
+      insert(realChildren, child)
+    end
+  end
+  return realChildren
```

Comment:
this should return both variants. Else this should be put in :getRealChildren() - there might be coding cases where you´d want children that are non trackable

Reply:
also what really means "real"?

Reply:
I don't recall changing Children.lua, but I'll look into this.

Reply:
Ah, OK -- what's happening here is that the Children interface is being used to add both non-component and component children to vessels. But getChildren and iterChildren are used by Economy, SystemMap and TrackingList in one way, and by Socket for components.

This blending caused problems with SystemMap thinking that components were docked at Stations. :D By telling getChildren and iterChildren to return only "real" (meaning not components) children, both Sockets and SystemMap (and the rest that use this interface) work correctly.

At some point this can be split into two interfaces if that delivers actual value in clarity/maintainability.